### PR TITLE
added personal keymap files, and sofle_rgb default maps,

### PR DESCRIPTION
**Pull request by DaneEvans**
, now with excess files properly culled, and the proper headers on files

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/sofle/keymaps/devdev/keymap.c [hunk 1]
```diff
@@ -0,0 +1,698 @@
...
+            } else {
+                if (keymap_config.swap_lctl_lgui) {
+                    unregister_mods(mod_config(MOD_LALT));
+                    unregister_code(KC_LEFT);
+                } else {
+                    unregister_mods(mod_config(MOD_LCTL));
+                    unregister_code(KC_LEFT);
+                }
+            }
+            break;
```
Comment (drashna) [inline keyboards/sofle/keymaps/devdev/keymap.c:544]: This should work as `LALT(KC_LEFT)` in the keymap itself.

File: keyboards/sofle/keymaps/rgb_default/config.h [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+/* Select hand configuration */
+
+///https://thomasbaart.nl/2018/12/01/reducing-firmware-size-in-qmk/
+
+
+#define MASTER_LEFT
+// #define MASTER_RIGHT
+// #define EE_HANDS
+
+#define SSD1306OLED
```
Comment (drashna) [inline keyboards/sofle/keymaps/rgb_default/config.h:30]:
```suggestion
```

File: users/devdev/config.h [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#pragma once
```
Comment (drashna) [inline users/devdev/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: users/devdev/glcdfont.c [hunk 4]
```diff
@@ -0,0 +1,465 @@
...
+0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+};
+#endif
+
+#ifdef CUSTOM_FONT
```
Comment (drashna) [inline users/devdev/glcdfont.c:236]:
```suggestion
#else
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/sofle/keymaps/devdev/keymap.c (hunk 1 above)
Comment (DaneEvans) [reply keyboards/sofle/keymaps/devdev/keymap.c]: Thanks, they were legacy from god knows where, I even chain several like that in my keymaps

File: users/devdev/glcdfont.c (hunk 4 above)
Comment (DaneEvans) [reply users/devdev/glcdfont.c]: *facepalm* how did I even?

**Review by drashna — APPROVED** [review]

File: keyboards/sofle/keymaps/rgb_default/rules.mk [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+LTO_ENABLE = yes
+
+
+OLED_DRIVER_ENABLE = yes  
+ENCODER_ENABLE = yes
+CONSOLE_ENABLE = yes
+EXTRAKEY_ENABLE = yes
+
+MOUSEKEY_ENABLE 	= yes  	# Mouse keys(+4700)
```
Comment (fauxpark) [inline keyboards/sofle/keymaps/rgb_default/rules.mk:11]:
```suggestion
MOUSEKEY_ENABLE = yes
EXTRAKEY_ENABLE = yes
CONSOLE_ENABLE = yes
RGBLIGHT_ENABLE = yes
ENCODER_ENABLE = yes
LTO_ENABLE = yes
OLED_DRIVER_ENABLE = yes
```

File: keyboards/sofle/keymaps/devdev/rules.mk [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+
+LTO_ENABLE = yes
+
+
+OLED_DRIVER_ENABLE = yes  
+ENCODER_ENABLE = yes
+CONSOLE_ENABLE = yes
+EXTRAKEY_ENABLE = yes
+
+MOUSEKEY_ENABLE 	= yes  	# Mouse keys(+4700)
```
Comment (fauxpark) [inline keyboards/sofle/keymaps/devdev/rules.mk:11]:
```suggestion
MOUSEKEY_ENABLE = yes
EXTRAKEY_ENABLE = yes
CONSOLE_ENABLE = yes
RGBLIGHT_ENABLE = yes
ENCODER_ENABLE = yes
LTO_ENABLE = yes
OLED_DRIVER_ENABLE = yes
```

File: keyboards/navi10/keymaps/devdev/rules.mk [hunk 7]
```diff
@@ -0,0 +1,4 @@
+TAP_DANCE_ENABLE = yes
+
+
+RGBLIGHT_ENABLE = yes       # Enable all of the lights
```
Comment (fauxpark) [inline keyboards/navi10/keymaps/devdev/rules.mk:4]:
```suggestion
RGBLIGHT_ENABLE = yes
TAP_DANCE_ENABLE = yes
```

File: keyboards/minimacro5/keymaps/devdev/rules.mk [hunk 8]
```diff
@@ -0,0 +1,4 @@
+
+
+TAP_DANCE_ENABLE = yes
+RGBLIGHT_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/minimacro5/keymaps/devdev/rules.mk:4]:
```suggestion
RGBLIGHT_ENABLE = yes
TAP_DANCE_ENABLE = yes
```

File: keyboards/crkbd/keymaps/devdev/rules.mk [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+#BACKLIGHT_ENABLE           = no  # Enable keyboard backlight functionality
+#MIDI_ENABLE                = no  # MIDI controls
+#AUDIO_ENABLE               = no  # Audio output on port C6
+#UNICODE_ENABLE             = no  # Unicode
+#BLUETOOTH_ENABLE           = no  # Enable Bluetooth with the Adafruit EZ-Key HID
+#SWAP_HANDS_ENABLE          = no  # Enable one-hand typing
+
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+#SLEEP_LED_ENABLE = no    # Breathing sleep LED during USB suspend
+
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/devdev/rules.mk:34]:
```suggestion
MOUSEKEY_ENABLE = yes
EXTRAKEY_ENABLE = yes
RGBLIGHT_ENABLE = yes
OLED_DRIVER_ENABLE = yes
```

File: keyboards/crkbd/keymaps/devdev/keymap.c [hunk 10]
```diff
@@ -0,0 +1,413 @@
...
+    _NUMPAD,
+	_SWITCH,
+    _MOVE,
+    
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+	
+		// colemak 
+	[_COLEMAK] = LAYOUT( \
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/devdev/keymap.c:42]: The backslashes here are not needed.

File: keyboards/sofle/keymaps/devdev/keymap.c [hunk 11]
```diff
@@ -0,0 +1,571 @@
...
+ * |------+------+------+------+------+------|                    |------+------+------+------+------+------|
+ * |LShift|   A  |   S  |   D  |   F  |   G  |-------.    ,-------|   H  |   J  |   K  |   L  |   ;  |  '   |
+ * |------+------+------+------+------+------| MUTE  |    |DISCORD|------+------+------+------+------+------|
+ * | LCTR |   Z  |   X  |   C  |   V  |   B  |-------|    |-------|   N  |   M  |   ,  |   .  |   /  |LShift|
+ * `-----------------------------------------/       /     \      \-----------------------------------------'
+ *            | Bspc | WIN  |LOWER | Enter| /Space  /       \Enter \  |SPACE |RAISE | RCTR | RAlt |
+ *            |      |      |      |      |/       /         \      \ |      |      |      |      |
+ *            `----------------------------------'           '------''---------------------------'
+ */
+  [_QWERTY] = LAYOUT( \
```
Comment (fauxpark) [inline keyboards/sofle/keymaps/devdev/keymap.c:104]: As above.

File: keyboards/sofle/keymaps/rgb_default/keymap.c [hunk 12]
```diff
@@ -0,0 +1,571 @@
...
+ * |------+------+------+------+------+------|                    |------+------+------+------+------+------|
+ * |LShift|   A  |   S  |   D  |   F  |   G  |-------.    ,-------|   H  |   J  |   K  |   L  |   ;  |  '   |
+ * |------+------+------+------+------+------| MUTE  |    |DISCORD|------+------+------+------+------+------|
+ * | LCTR |   Z  |   X  |   C  |   V  |   B  |-------|    |-------|   N  |   M  |   ,  |   .  |   /  |LShift|
+ * `-----------------------------------------/       /     \      \-----------------------------------------'
+ *            | Bspc | WIN  |LOWER | Enter| /Space  /       \Enter \  |SPACE |RAISE | RCTR | RAlt |
+ *            |      |      |      |      |/       /         \      \ |      |      |      |      |
+ *            `----------------------------------'           '------''---------------------------'
+ */
+  [_QWERTY] = LAYOUT( \
```
Comment (fauxpark) [inline keyboards/sofle/keymaps/rgb_default/keymap.c:104]: etc.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by DaneEvans** [comment]
@fauxpark poke

**Review by fauxpark — APPROVED** [review]

File: keyboards/crkbd/keymaps/devdev/keymap.c [hunk 13]
```diff
@@ -0,0 +1,410 @@
...
+	layer_numpad_lights, 
+	layer_numpad_rh_lights,
+	layer_move_lights,
+	layer_switcher_lights  // Overrides other layers
+);
+
+void keyboard_post_init_user(void) {
+    // Enable the LED layers
+    rgblight_layers = my_rgb_layers;
+	rgblight_mode(10);// haven't found a way to set this in a more useful way
```
Comment (drashna) [inline keyboards/crkbd/keymaps/devdev/keymap.c:239]:
```suggestion
	rgblight_mode_noeeprom(10);// haven't found a way to set this in a more useful way 
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**